File: src/main/java/net/gegy1000/terrarium/server/map/source/raster/RasterSource.java
package net.gegy1000.terrarium.server.map.source.raster;

import net.gegy1000.terrarium.server.util.Coordinate;

public interface RasterSource<T> {
    T get(Coordinate coordinate);

    void sampleArea(T[] data, Coordinate coordinate, Coordinate size);
}

File: src/main/kotlin/net/gegy1000/terrarium/server/map/source/raster/RasterSource.kt
package net.gegy1000.terrarium.server.map.source.raster

import net.gegy1000.terrarium.server.util.Coordinate

interface RasterSource<V> {
    fun get(coordinate: Coordinate): V

    fun sampleArea(result: Array<V>, minimumCoordinate: Coordinate, maximumCoordinate: Coordinate)
}
